File: LintCode/001/main.cpp
class Solution {
public:
    /*
     * @param : An integer
     * @param : An integer
     * @return: The sum of a and b
     */
    int aplusb(int a, int b) {
        // write your code here
        while (a&b) {
            int aa=a;
            int bb=b;
            a=aa^bb;
            b=(aa&bb)<<1;
        }
        return a^b;
    }
};

File: LintCode/001/main.py
class Solution:
    """
    @param a: An integer
    @param b: An integer
    @return: The sum of a and b
    """

    def aplusb(self, a, b):
        # write your code here, try to do it without arithmetic operators.
        limit = 0xfffffffff
        while b != 0:
            a, b = (a ^ b) & limit, (a & b) << 1
        return a if a & 1 << 32 == 0 else a | (~limit)
